logstash/examples/default/values.yaml

```diff
@@ -0,0 +1,17 @@
+extraEnvs:
+  - name: XPACK_MONITORING_ENABLED
+    value: "false"
+
+#logstashConfig:
+#  logstash.yml: |
+#    xpack.monitoring.enabled: false
```

Comment:
Note that commenting `XPACK_MONITORING_ENABLED` environment variable and uncommenting `logstash.yml` config is currently failing.

```
cat << EOF > values.yaml
#extraEnvs:
#  - name: XPACK_MONITORING_ENABLED
#    value: "false"
logstashConfig:
  logstash.yml: |
    xpack.monitoring.enabled: false
EOF

# this is failing
helm install --values values.yaml logstash

kubectl logs logstash-logstash-0
2019/10/16 18:40:55 Setting 'node.name' from environment.
2019/10/16 18:40:55 error: open /usr/share/logstash/config/logstash.yml: read-only file system
```

This seem related to [env2yaml](https://github.com/elastic/logstash/blob/master/docker/data/logstash/env2yaml/env2yaml.go) trying to edit the mounted `/usr/share/logstash/config/logstash.yml` to add `node.name` environment variable that I set in [container spec](https://github.com/elastic/helm-charts/pull/333/files#diff-e1831a0b9e9999a7d2f6f2276f9fd95aR149-R152). This part comes from Elasticsearch chart copy/paste so I guess I should remove it, however AFAIK [logstashconfig Volume](https://github.com/elastic/helm-charts/pull/333/files#diff-e1831a0b9e9999a7d2f6f2276f9fd95aR171-R173) shouldn't be mounted read-only, so I don't know why it's not writable by `env2yaml`.

Reply:
So as configmap are bind-mounted, it seems they can't be edited even if they are seen as read-write in the pod description:

```
kubectl describe pod helm-logstash-default-logstash-0
...
Containers:
  logstash:
...
    Mounts:
      /usr/share/logstash/config/logstash.yml from logstashconfig (rw,path="logstash.yml")
...
Volumes:
...
  logstashconfig:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      helm-logstash-default-logstash-config
    Optional:  false
...
```

In summary, you can't mix configuration using configmap and environment variables for `logstash.yml` (details in https://github.com/elastic/helm-charts/pull/333/commits/bcb7af6eb128a54c1c3e61124ff6a1f0b0709aaa).